### DISCUSSION TITLE
Help with Predictions

### DISCUSSION THREAD

**Discussion by connorwforman** [talmolab/sleap#1174] [Help!]
Hello! New to sleap here. I'm trying to use the training feature to predict new instances, I may be using it completely wrong. 

I have a test here, a ball rolling across the screen with ~17 frames labeled at different points in its motion. 
![image](https://user-images.githubusercontent.com/35116511/218343702-d628a135-b90a-4385-9670-3104175b79b9.png)

I'm not even sure if this is what it is used for, but after training is complete, it says it made 0 predictions. I would like it to predict the location of the instance between the labeled key frames to make the tracking of the ball visually smoother. Similar to key frame animation. 

Am I using the program wrong, is that not what training is for? Is there another way to do this? Should I be using a different program?

Any help or advice is appreciated!

**Comment by roomrys** [comment]
Hi @connorwforman,

SLEAP should be able to do this. Can you try the **top-down** model with the following settings:

**Centroid configuration**:
- Reduce the input scaling / increase the max stride until the Receptive Field (RF), the blue box drawn on top of the sample frame in the configuration, contains the entire ball. We should be able to see the point we want to track within the RF. Since the point you are tracking (the centroid) is more easily located by viewing the entire ball, it would be helpful to include the entire ball in the RF.

**Centered-instance configuration**:
- Set the epochs to train for only 1, we can essentially skip this step since we are really only interested in finding the centroid which should be accomplished by the centroid model above. If you notice that the predictions are slightly off from your labeled point, then increasing the number of epochs for this model will help correct that.

You can also try the Single Instance model, but you should ensure that the RF contains the entire ball. Bottom-up is not supported for single node skeletons at the moment. Top-down is the most memory efficient, especially if the object you want to track is relatively small compared to the rest of the frame.

Let us know how it goes! If you are still having trouble, would you mind posting your training settings here? A follow-up question is whether the ball grows larger/smaller as it rolls around the screen? - which would affect what we suggest doing.

Thanks,
Liezl
